src/document-converter.ts

```diff
@@ -803,12 +829,17 @@ export class DocumentConverter {
             return;
           }
           const [callPath] = assignmentPath.replace(jsc.callExpression(
-              jsc.identifier(setterName), [assignmentPath.node.right]));
+              jsc.identifier('IMPORT_REFERENCE_REPLACEMENT_FAILED'),
```

Comment:
Handle this error (throw an exception? log a warning?) instead of replacing the node with a dummy identifier

Reply:
The identifier here should always be replaced but I have to supply some name for it at this point so I figured it would be good to give it a name that made it obvious that it shouldn't be in the output.

Reply:
(Inserting this identifier into the program at this point is not the potential error that this identifier's name is meant to bring attention to.)

Reply:
But does this get replaced again later? Otherwise, aren't you silently adding an invalid identifier to your code?

If this is absolutely required, at the very least please add some documentation to why this is okay & where it is handled later in the flow.

Reply:
> But does this get replaced again later?

Yes, this identifier is [set as the AST node (`path`) for this `ImportReference`](https://github.com/Polymer/polymer-modulizer/pull/165/files#diff-4c7c2c6e37f696bf962a9f9da25cf010R840), so it should always be replaced later, during import aliasing. However, this particular point in the modulizer code is where some gets/sets are transformed into calls to modulizer-created and imported getter/setter functions: I'm not really sure how it determines when to do this, but modulizer seems to detect if you were trying to do something like `Scope.x = y;` and changes it to something like `import { setX } from "./Scope.js"; /* ... */ setX(y);`, if appropriate. So, AFAICT, I have to produce an AST node for the 'function' part (`setX`) of the 'function call' node (the whole `setX(y)`) even though this node is also the one that will be replaced during aliasing and shouldn't ever appear in the output.

You're totally right about this needing some documentation. I'll add a comment here to explain why this is happening.

Reply:
I added the comment, let me know if you think it seems clear enough.